test/cbmc/proofs/DNS/DNSgetHostByName/DNSgetHostByName_harness.c

```diff
@@ -94,32 +98,38 @@ uint32_t DNS_SendRequest( Socket_t xDNSSocket,
 * We stub out this function which returned a dns_buffer filled with random data
 *
 ****************************************************************/
-void DNS_ReadReply( Socket_t xDNSSocket,
+BaseType_t DNS_ReadReply( const ConstSocket_t xDNSSocket,
                     struct freertos_sockaddr * xAddress,
                     struct xDNSBuffer * pxDNSBuf )
 {
+    BaseType_t ret;
     int len;
 
+    __CPROVER_assume( ( len > sizeof( DNSMessage_t ) ) && ( len < CBMC_MAX_OBJECT_SIZE ) );
+
     pxDNSBuf->pucPayloadBuffer = safeMalloc( len );
 
     pxDNSBuf->uxPayloadLength = len;
 
-    __CPROVER_assume( len < CBMC_MAX_OBJECT_SIZE );
     __CPROVER_assume( pxDNSBuf->pucPayloadBuffer != NULL );
 
-    __CPROVER_havoc_slice( pxDNSBuf->pucPayloadBuffer, pxDNSBuf->uxPayloadSize );
+    __CPROVER_havoc_slice( pxDNSBuf->pucPayloadBuffer, pxDNSBuf->uxPayloadLength );
+
+    return ret;
 }
 
 
 void DNS_CloseSocket( Socket_t xDNSSocket )
 {
+    __CPROVER_assert( xDNSSocket != NULL, "The xDNSSocket cannot be NULL." );
```

Comment:
Why not free the socket which we are mallocing in DNS_CreateSocket?

Reply:
fixed